### ISSUE TITLE
ServiceController createLoadBalancerIfNeeded makes unneeded calls to cloud provider

### ISSUE THREAD

**Issue by pmorie** [kubernetes/kubernetes#30700]
The service controller can make an unnecessary call to the cloud provider to get a load balancer for the service, even if one is not defined, if that service is updated: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/service/servicecontroller.go#L263

... this can surface to the user with an event that makes it look like:
1.  The service controller incorrectly attempted to create a load balancer
2.  That incorrect load balancer creation failed

...if the cloud account isn't permissioned to get load balancers.

Example:

```
2016-08-15T23:28:29Z   2016-08-15T23:28:29Z   1         cakephp-mysql-example   Service             Warning   CreatingLoadBalancerFailed   {service-controller }   Error creating load balancer (will retry): Error getting LB for service dakinitest/cakephp-mysql-example: AccessDenied: User: :user/cloud_provider is not authorized to perform: elasticloadbalancing:DescribeLoadBalancers
```

The fix should be to:
1. Not make the `GetLoadBalancer` call if it is unnecessary (ie, if the service was updated, but never had type=LoadBalancer
2. If the behavior is that the controller is deleting the load balancer, the event associated with a failure should not be `CreatingLoadBalancerFailed`

**Comment by pmorie** [comment]
cc @kubernetes/rh-cluster-infra

**Comment by AdoHe** [comment]
@pwittrock why k8s-merge-robot add `component/kubectl` label to this one?

**Comment by pwittrock** [comment]
@AdoHe Because our machine-learning robot has a long ways to go

**Comment by pmorie** [comment]
I'll fix this after master opens for 1.5 if no one else takes it first.

**Comment by icambridge** [comment]
This issue can be closed?
